React/Recipes/Session Handling.md

```diff
@@ -279,15 +279,19 @@ const AuthRedirect: FC<IAuthRedirectProps> = ({ to, ifFound, condition }) => {
     const shouldRedirect = (ifFound && data) || (!ifFound && !data);
 
     if (shouldRedirect) {
-      router.push(to);
+      push(to);
     }
-  }, [data, isValidating, error, to, condition, ifFound, router]);
+  }, 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  [data, isValidating, error, to, ifFound]);
 
   // this component renders nothing since it is only used to redirect if needed.
   return null;
 };
 ```
 
+> **Note** we added `eslint-disable-next-line react-hooks/exhaustive-deps` to the `useEffect` hook. This is because we don't want to rerender the component if `condition` and `push` are changed. We are aware this is a dangerous thing to do, but in this case, it is a necessary evil. This issue will be resolved when React add finish `useEffectEvent` hook. More about this can be found [here](https://react.dev/learn/separating-events-from-effects#reading-latest-props-and-state-with-effect-events)
```

Comment:
```suggestion
> **Note** we added `eslint-disable-next-line react-hooks/exhaustive-deps` to the `useEffect` hook. This is because we don't want to rerender the component if `condition` and `push` are changed. We are aware this is a dangerous thing to do, but in this case, it is a necessary evil. This issue will be resolved when React finishes `useEffectEvent` hook. More about this can be found [here](https://react.dev/learn/separating-events-from-effects#reading-latest-props-and-state-with-effect-events)
```